engine/src/animation.rs

```diff
@@ -0,0 +1,334 @@
+use crate::assets::animation::*;
+use crate::prelude::*;
+
+pub struct AnimationPlugin;
+
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(sprite_animation_update.in_set(GameTickSet::Pre));
+    }
+}
+
+#[derive(Component)]
+pub struct SpriteAnimationPlayer {
+    animation: Handle<SpriteAnimation>,
+    state: SpriteAnimationPlayerState,
+}
+
+enum SpriteAnimationPlayerState {
+    Stopped,
+    PendingPlay {
+        settings: PlaySettings,
+    },
+    Playing {
+        settings: PlaySettings,
+        scripts: PlayScripts,
+        state: PlayState,
+    },
+}
+
+#[derive(Debug, Clone)]
+struct PlaySettings {
+    paused: PauseMode,
```

Comment:
It seems like this is kind of being used as a pause state, would it make sense to add this to `SpriteAnimationPlayerState`?